versioned/tiered/rocksdb/src/main/java/com/dremio/nessie/versioned/store/rocksdb/RocksBaseValue.java

```diff
@@ -0,0 +1,192 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.dremio.nessie.versioned.store.rocksdb;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+import com.dremio.nessie.tiered.builder.BaseValue;
+import com.dremio.nessie.versioned.Key;
+import com.dremio.nessie.versioned.store.Entity;
+import com.dremio.nessie.versioned.store.Id;
+import com.google.common.base.Preconditions;
+import com.google.protobuf.ByteString;
+
+/**
+ * An implementation of @{BaseValue} used for ConditionExpression and UpdateExpression evaluation.
+ * @param <C> Specialization of a specific BaseValue interface.
+ */
+abstract class RocksBaseValue<C extends BaseValue<C>> implements BaseValue<C> {
+
+  public static final String ID = "id";
+
+  private Id id;
+  private long dt;
+
+  RocksBaseValue() {
+  }
+
+  @SuppressWarnings("unchecked")
+  @Override
+  public C id(Id id) {
+    this.id = id;
+    return (C) this;
+  }
+
+  @SuppressWarnings("unchecked")
+  @Override
+  public C dt(long dt) {
+    this.dt = dt;
+    return (C) this;
+  }
+
+  Id getId() {
+    return id;
+  }
+
+  /**
+   * Converts a Stream of Ids into a List Entity.
+   * @param idStream the stream of Id to convert.
+   * @return the List Entity.
+   */
+  Entity toEntity(Stream<Id> idStream) {
+    List<Id> idList = idStream.collect(Collectors.toList());
+    List<Entity> idsAsEntity = new ArrayList<>();
+    for (Id idElement : idList) {
+      idsAsEntity.add(idElement.toEntity());
+    }
+    return Entity.ofList(idsAsEntity);
+  }
+
+  /**
+   * Retrieves an Id at 'position' in a Stream of Ids as an Entity.
+   * @param idStream the stream of Id to convert.
+   * @param position the element in the Stream to retrieve.
+   * @return the List Entity.
+   */
+  Entity toEntity(Stream<Id> idStream, int position) {
+    return idStream.collect(Collectors.toList()).get(position).toEntity();
+  }
+
+  /**
+   * Evaluates if the stream of Id meets the Condition Function.
+   * @param function the function of the ConditionExpression to evaluate.
+   * @param stream The Ids to evaluate against the function
+   * @return true if the condition is met
+   */
+  boolean evaluateStream(Function function, Stream<Id> stream) {
+    // EQUALS will either compare a specified position or the whole stream as a List.
+    final List<String> path = Evaluator.splitPath(function.getPath());
+    final String segment = path.get(0);
+    if (path.size() == 1) {
+      if (function.getOperator().equals(Function.EQUALS)) {
+        // Remove enclosing brackets.
+        final List<String> arguments = Arrays.asList(segment.split("[()]"));
+        if (arguments.size() == 1) { // compare complete list
```

Comment:
I don't understand the logic for the split of 1 arg vs. 2 for equals, can this be commented?

Reply:
I refactored this using ExpressionPath. I hope it is more obvious.